test_client_context.py

```diff
@@ -0,0 +1,30 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+from client_context import ClientContext
+from fh_state_base_holder import FeatureStateBaseHolder
+
+
+class TestClientContext(TestCase):
+
+    @patch('featurehub_repository.FeatureHubRepository')
+    @patch('edge_service.EdgeService')
+    def test_get_number(self, mock_repo, edge_service):
+        var = FeatureStateBaseHolder({'id': '123', 'key': 'FEATURE_TITLE_TO_UPPERCASE',
+                                      'l': True, 'version': 1, 'type': 'NUMBER', 'value': 3, 'strategies': []},)
+
+        mock_repo.feature.return_value = var
+        client_context = ClientContext(mock_repo, edge_service)
+        result = client_context.get_number("bla")
+        self.assertEqual(result, 3)
+
+    @patch('featurehub_repository.FeatureHubRepository')
+    @patch('edge_service.EdgeService')
+    def test_get_number_when_none(self, mock_repo, mock_edge):
+        var = FeatureStateBaseHolder({'id': '123', 'key': 'FEATURE_TITLE_TO_UPPERCASE',
+                                      'l': True, 'version': 1, 'type': 'BOOLEAN', 'value': 'true', 'strategies': []},)
+
+        mock_repo.feature.return_value = var
+        client_context = ClientContext(mock_repo, mock_edge)
```

Comment:
this is not quite right, it needs to be mock_repo.feature.return_value = None - because that what is the case you are testing for

Reply:
if I do this, it means function feature() returns None, but here I am testing totally different thing - get_number() function in case there is no number, but for example boolean instead

Reply:
Right, I resolved a thing above where you said the get_number for a non existent feature worked and didn't blow up, so I presumed this was the test. Are you able to add that test?